chromium_src/net/url_request/url_request_job.cc

```diff
@@ -0,0 +1,25 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "net/url_request/url_request_job.h"
+
+#define ComputeReferrerForPolicy                                              \
+  ComputeReferrerForPolicy(                                                   \
+      ReferrerPolicy policy, const GURL& original_referrer,                   \
+      const GURL& destination, bool* same_origin_out_for_metrics) {           \
+    if (base::EndsWith(original_referrer.host_piece(), ".onion",              \
+                       base::CompareCase::INSENSITIVE_ASCII) &&               \
+        !url::Origin::Create(original_referrer)                               \
+             .IsSameOriginWith(url::Origin::Create(destination))) {           \
```

Comment:
there is a `IsSameOriginWith(GURL, GURL)` version which does `Origin::Create` for you.
https://github.com/chromium/chromium/blob/97.0.4685.0/url/origin.cc#L395

Reply:
Good idea. Done in latest revision.